yarn-project/pxe/src/kernel_prover/kernel_prover.ts

```diff
@@ -126,21 +136,10 @@ export class KernelProver {
       firstIteration = false;
     }
 
-    let previousVkMembershipWitness = await this.oracle.getVkMembershipWitness(output.verificationKey);
-    let previousKernelData = new PrivateKernelData(
-      output.publicInputs,
-      output.proof,
-      output.verificationKey,
-      Number(previousVkMembershipWitness.leafIndex),
-      assertLength<Fr, typeof VK_TREE_HEIGHT>(previousVkMembershipWitness.siblingPath, VK_TREE_HEIGHT),
-    );
-
-    output = await this.proofCreator.createProofReset(
-      await buildPrivateKernelResetInputs(previousKernelData, noteHashLeafIndexMap, this.oracle),
-    );
+    output = await this.runReset(executionStack, output, noteHashLeafIndexMap, noteHashNullifierCounterMap);
```

Comment:
It's possible that we don't have to run reset here!

Reply:
I can add a check that we have no more validation requests and no transient note hashes or nullifiers!